### DISCUSSION TITLE
Exploratory Data Analysis

### DISCUSSION THREAD

**Discussion by kkms51** [teamalgoritma/community#904] [Q&A] [answered]
Saya ingin bertanya mengenai how to handle missing values and inconsistency datas. Itu apa saja yang mesti saya lakukan? Apakah adacara lain, selain bfill, ffill, mean, dll. Ketika data pada cell tersebut tidak bisa saya hapus karena sangat penting. Terima kasih

**Comment by adizaen** [comment, marked as answer]
Dear @kkms51 

Terima kasih atas pertanyaannya Pak. 
Untuk treatment missing values sebenarnya ada beberapa cara umum yang bisa digunakan. Antara lain:

1. Hapus baris atau kolom: Menggunakan metode `dropna()`
2. Replace NA dengan nilai `mean`, `median`, dll
3. Tetap mempertahankan data kita

Jika datanya tidak dihapus, ada proses imputasi untuk mengisi value tersebut Pak. Bapak bisa melakukan imputasi terhadap data yang mengandung missing value, menggunakan metode `.fillna()`. Berikut ada beberapa tips untuk imputasi.

**Untuk kolom categorical:**
- Isi menggunakan nilai lainnya seperti "other" atau "missing"
- Isi menggunakan nilai terbanyak yg muncul (mode)

**Untuk kolom numerical:**
- Isi menggunakan nilai tengah seperti `mean` atau `median`
- Isi dengan nilai 0

**Untuk kolom datetime:**
- Menggunakan metode `bfill` : melakukan imputasi dari bawah ke atas
- Menggunakan metode `ffill` : melakukan imputasi dari atas ke bawah

Semoga membantu Pak.

Best regards,
**Team Algoritma**

**Comment by saskia-dwi-ulfah** [comment]
Dear @kkms51 

Terima kasih telah bertanya melalui Github discussions Algoritma. Untuk menjaga agar forum diskusi berjalan kondusif, karena tidak ada respon lanjutan setelah 7 hari dari respon terakhir, kami izin menandai pertanyaan sudah terjawab dan menutup diskusi ini. Semoga dapat dipahami dan terima kasih.

Best regards,
Team Algoritma
